### ISSUE TITLE
Modify a process instance by terminating given element instances

### ISSUE THREAD

**Issue by saig0** [camunda/zeebe#9690]
## Description

We want to support the basic modification by terminating given element instances. We want to terminate the element instance for each terminate instruction. The element instances should be terminated by writing a `terminating` and `terminated` event for the given element instances. All related resources, like event subscriptions, should be deleted, similar to a regular termination of an element.

For example:
![image](https://user-images.githubusercontent.com/4305769/177271007-b52b9c99-0456-4e37-979b-c756cd3d699b.png)

* for each terminate instruction:
  * write a `terminating` and `terminated` event
  * delete event subscriptions if exist (i.e. timers, message subscriptions)
  * cancel the job if exists (e.g. for service tasks)
  * resolve incident if exists
* verify that 
  * the process instance terminates the given element instances
  * the process instance can't complete the given element instances after termination
  * the process instance is completed when the other active element instances are completed

Out of scope:
* termination of container elements (e.g. subprocess, call activity)
* termination of all element instances of a flow scope

Blocked by 
- #9639
